### DISCUSSION TITLE
useQuery remove() question

### DISCUSSION THREAD

**Discussion by Jeffdude** [TanStack/query#1807] [Q&A]
I have a page which is displaying the data of a "submission" obtained through two useQuery calls ([Called here](https://github.com/JMKRIDE-USA/AmbassadorSite-frontend/blob/master/src/components/submission-display.js#L310-L311), bottom of my code stack is [here](https://github.com/JMKRIDE-USA/AmbassadorSite-frontend/blob/master/src/modules/data.js#L24)). I also have a mutation button to delete the submission, [which calls remove() on both queries, and redirects to /](https://github.com/JMKRIDE-USA/AmbassadorSite-frontend/blob/master/src/components/submission-display.js#L314-L319).

After deletion, I'm still seeing errors in the logs from these queries that are still trying to fetch. Even on "/", a different navigation screen. Is there something I need to do with the queryClient object to fully stop these queries from running again?

Thanks for the support.

**Comment by TkDodo** [comment]
I can take a look if you can reproduce this in a codesandbox. I’ve never used remove. I think calling remove will still notify active observers.
Instead, you can set `cacheTime: 0` so that the query will be removed if no observers are active.
